lib/Backend/GlobOptBailOut.cpp

```diff
@@ -15,52 +15,258 @@ GlobOpt::CaptureValue(BasicBlock *block, StackSym * stackSym, Value * value, Bai
 
-        if (valueInfo->GetSymStore() == nullptr && !valueInfo->HasIntConstantValue())
+        FOREACH_BITSET_IN_SPARSEBV(symId, tempBv)
         {
-            continue;
+            HashBucket<Sym*, Value*> * bucket = block->globOptData.symToValueMap->GetBucket(symId);
+            StackSym * stackSym = bucket->value->AsStackSym();
+            value = block->globOptData.capturedArgs->Test(symId) ? FindValue(stackSym) : bucket->element;
+            valueInfo = value->GetValueInfo();
+
+            int intConstantValue;
+            if (valueInfo->TryGetIntConstantValue(&intConstantValue))
+            {
+                BailoutConstantValue constValue;
+                constValue.InitIntConstValue(intConstantValue);
+                bailOutConstValuesIter.InsertNodeBefore(this->func->m_alloc, stackSym, constValue);
+            }
+            else if (valueInfo->IsVarConstant())
+            {
+                BailoutConstantValue constValue;
+                constValue.InitVarConstValue(valueInfo->AsVarConstant()->VarValue());
+                bailOutConstValuesIter.InsertNodeBefore(this->func->m_alloc, stackSym, constValue);
+            }
+            else
+            {
+                StackSym* copyPropSym = this->GetCopyPropSym(block, stackSym, value);
+                if (copyPropSym)
+                {
+                    bailOutCopySymsIter.InsertNodeBefore(this->func->m_alloc, stackSym, copyPropSym);
+                }
+            }
         }
-        Sym * sym = bucket.value;
-        if (sym == nullptr || !sym->IsStackSym() || !sym->AsStackSym()->HasByteCodeRegSlot())
+        NEXT_BITSET_IN_SPARSEBV
+    }
+    else
+    {
+        // capture bailout values incrementally
+
+        CapturedValues * currCapturedValues = block->globOptData.capturedValues;
+        SListBase<ConstantStackSymValue>::Iterator iterConst(currCapturedValues ? &currCapturedValues->constantValues : nullptr);
+        SListBase<CopyPropSyms>::Iterator iterCopyPropSym(currCapturedValues ? &currCapturedValues->copyPropSyms : nullptr);
+        bool hasConstValue = currCapturedValues ? iterConst.Next() : false;
+        bool hasCopyPropSym = currCapturedValues ? iterCopyPropSym.Next() : false;
+
+        tempBv->Or(block->globOptData.changedSyms, block->globOptData.capturedArgs);
+
+        tempBv->Set(Js::Constants::InvalidSymID);
+
+        FOREACH_BITSET_IN_SPARSEBV(symId, tempBv)
         {
-            continue;
+            Sym * sym = hasConstValue ? iterConst.Data().Key() : nullptr;
+            Value * val = nullptr;
+            HashBucket<Sym *, Value *> * symIdBucket = nullptr;
+            bool symIdHandled = false;
+
+            // copy unchanged sym to new capturedValues
+            while (sym && sym->m_id < symId)
+            {
+                Assert(sym->IsStackSym());
+                if (!sym->AsStackSym()->HasArgSlotNum() || block->globOptData.capturedArgs->Test(sym->m_id))
```

Comment:
This should always be false, otherwise the sym would be in the changedSym bit vector.